src/main/java/uk/ac/ebi/embl/api/validation/submission/SubmissionValidationPlan.java

```diff
@@ -151,149 +160,185 @@ public void execute() throws ValidationEngineException {
+			return result;
 		String fileName=null;
-
-		try
-		{
+		try {
 			check = new FastaFileValidationCheck(options);
-			for(SubmissionFile fastaFile:options.submissionFiles.get().getFiles(FileType.FASTA))
-			{
+			for (SubmissionFile fastaFile : options.submissionFiles.get().getFiles(FileType.FASTA)) {
 				fileName= fastaFile.getFile().getName();
-				if(sequenceDB!=null)
+				if (sequenceDB != null)
 					check.setSequenceDB(sequenceDB);
-				if(contigDB!=null)
+				if (contigDB != null)
 					check.setContigDB(contigDB);
-				if(!check.check(fastaFile))
-					throwValidationCheckException(FileType.FASTA,fastaFile);
-				else if(!options.isRemote)
-				     flagValidation(FileType.FASTA);
-
+				result = check.check(fastaFile);
+				if (!result.isValid()) {
+					result.setDefaultOrigin(new DefaultOrigin(getValidationErrorMessage(FileType.FASTA, fastaFile, check)));
+					return result;
+				}
 			}
-		}catch(Exception e)
-		{
+			if (!options.isRemote)
+				flagValidation(FileType.FASTA);
+		} catch(ValidationEngineException e) {
 			throwValidationEngineException(FileType.FASTA,e,fileName);
+		}catch(IOException e)
+		{
+			throw new ValidationEngineException(e);
 		}
+		return result;
 	}
 
-	private void validateFlatfile() throws ValidationEngineException
+	private ValidationResult validateFlatfile() throws ValidationEngineException
 	{
+		ValidationResult result = new ValidationResult();
 		if(options.processDir.isPresent()&&Files.exists(Paths.get(String.format("%s%s%s",options.processDir.get(),File.separator,flatfileFlagFileName))))
-			return;
+			return result;
 		String fileName=null;
-
 		try
 		{
-			check = new FlatfileFileValidationCheck(options);
-			for(SubmissionFile flatfile:options.submissionFiles.get().getFiles(FileType.FLATFILE))
-			{
-				fileName= flatfile.getFile().getName();
-				if(sequenceDB!=null)
-					check.setSequenceDB(sequenceDB);
-				if(contigDB!=null)
-					check.setContigDB(contigDB);
-				if(!check.check(flatfile))
-					throwValidationCheckException(FileType.FLATFILE,flatfile);
-				else if(!options.isRemote)
-				     flagValidation(FileType.FLATFILE);
+		check = new FlatfileFileValidationCheck(options);
+		for(SubmissionFile flatfile:options.submissionFiles.get().getFiles(FileType.FLATFILE))
+		{
+			fileName= flatfile.getFile().getName();
+			if(sequenceDB!=null)
+				check.setSequenceDB(sequenceDB);
+			if(contigDB!=null)
+				check.setContigDB(contigDB);
+			result = check.check(flatfile);
+			if(!result.isValid()) {
+				result.setDefaultOrigin(new DefaultOrigin(getValidationErrorMessage(FileType.FLATFILE,flatfile, check)));
+				return result;
 			}
+		}
+			if (!options.isRemote)
+				flagValidation(FileType.FLATFILE);
+		}catch(ValidationEngineException e){
+			throwValidationEngineException(FileType.FLATFILE,e,fileName);
 		}catch(Exception e)
 		{
-			throwValidationEngineException(FileType.FLATFILE,e,fileName);
+			throw new ValidationEngineException();
 		}
```

Comment:
Now sure why we don't just do:

catch(Exception e)
{
throwValidationEngineException(FileType. FLATFILE,e,"master.dat");
}